enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -566,22 +591,43 @@ def can_redeem(self, request, enterprise_customer_uuid):
                         "reason": reason,
                         "policy_uuids": [policy.uuid for policy in policies],
                     })
+
+            # Select one redeemable policy (i.e. use the "policy resolver") and serialize it for output.
+            resolved_policy = None
             if redeemable_policies:
                 resolved_policy = SubsidyAccessPolicy.resolve_policy(redeemable_policies)
-                serialized_policy = serializers.SubsidyAccessPolicyRedeemableResponseSerializer(resolved_policy).data
 
+            # Determine if the learner has already redeemed the requested content_key.
             has_successful_redemption = any(
                 redemption['state'] == TransactionStateChoices.COMMITTED
                 for redemption in redemptions
             )
-            can_redeem_for_content_response = {
+
+            # Determine the price for content for display purposes only.
+            try:
+                content_metadata = self.subsidy_client.get_subsidy_content_data(enterprise_customer_uuid, content_key)
+                # Note that the "content_price" key is guaranteed to exist, but the value may be None.  Also, the
+                # upstream field type is a string representing a decimal dollar.
+                list_price_decimal_dollars_str = content_metadata["content_price"]
+                list_price_integer_cents = int(float(list_price_decimal_dollars_str) * 100)
+            except requests.exceptions.HTTPError:
+                # No need to record a failure reason here because evaluate_policies() -> policy.can_redeem() already
+                # caught any problematic or non-existent content keys, and provided a reason it could not be redeemed.
+                # Just set a None content_price and move along.
+                list_price_integer_cents = None
+
+            element_response = {
                 "content_key": content_key,
+                "list_price": list_price_integer_cents,
                 "redemptions": redemptions,
                 "has_successful_redemption": has_successful_redemption,
-                "redeemable_subsidy_access_policy": serialized_policy,
-                "can_redeem": bool(serialized_policy),
+                "redeemable_subsidy_access_policy": resolved_policy,
+                "can_redeem": bool(resolved_policy),
                 "reasons": reasons,
             }
-            response.append(can_redeem_for_content_response)
+            element_responses.append(element_response)
 
-        return Response(response, status=status.HTTP_200_OK)
+        response_serializer = serializers.SubsidyAccessPolicyCanRedeemResponseSerializer(
+            {"results": element_responses}
```

Comment:
This is technically a breaking change to update the response from a list to a dict with a `results` list, as the client (learner portal MFE) currently expects a list to be returned. If we change this, it will likely result in a JavaScript error when parsing its response in the learner portal course page on stage.

A couple links to learner portal source where we parse the `can-redeem` response:
* https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/2e8ebf2beac823651019338352dd2841d7b860c2/src/components/course/data/hooks.jsx#L552-L557
* https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/2e8ebf2beac823651019338352dd2841d7b860c2/src/components/course/data/hooks.jsx#L567

![image](https://github.com/openedx/enterprise-access/assets/2828721/2bec377b-5be2-4bd3-91ac-1ab9b4d2d74b)

A few things to consider: 
1. be OK with potentially breaking stage learner portal course page until it is updated to reflect this change shortly after (will not impact production).
2. version the endpoint.
3. update learner portal course page to handle both style of responses and then remove the old way.
4. something else?

Reply:
Good catch, I forgot to mention this.  I re-worked the endpoint to revert it back to original behavior, with ample code comments to explain why we chose to deviate from the industry norm.  Also, I re-worked it in a way that the redoc page looks accurate:
![image](https://github.com/openedx/enterprise-access/assets/85151/a29bd560-e894-4c12-8a3a-b8eec961ef45)